SI/Sem.05/Pract.05/Task06.cpp

```diff
@@ -0,0 +1,43 @@
+#include <iostream>
+
+int main() {
+	const int MAX_LENGTH = 100;
+	int arr[MAX_LENGTH];
+	std::cout << "enter size (<=" << MAX_LENGTH << "):" << std::endl;
+	int n;
+	std::cin >> n;
+	if (n > MAX_LENGTH) {
+		std::cout << "size must be <=" << MAX_LENGTH << std::endl;
+		return 0;
+	}
+	std::cout << "enter elements:" << std::endl;
+	for (int i = 0; i < n; i++) {
+		int element;
+		std::cin >> element;
+		arr[i] = element;
+	}
+
+	//sort
+	int i, j;
+	for (i = 0; i < n - 1; i++) { 
+		for (j = 0; j < n - i - 1; j++) { 
+			if (arr[j] > arr[j + 1]) {
+				int cur = arr[j];
+				arr[j] = arr[j + 1];
+				arr[j + 1] = cur;
+			}
+		}
+	}
+
+	//sort by rule
+	for (int i = 1; i < n; i+=2) {
+		int cur = arr[i];
+		arr[i] = arr[i - 1];
+		arr[i - 1] = cur;
+	}
```

Comment:
The solution can be done without sorting, but just with one loop that iterates through the array and if two numbers(one after another) are not in the correct order - swap